src/ac_flag.c

```diff
@@ -0,0 +1,407 @@
+
+    if (mMsg_Check_main_wait(msg_p) != FALSE) {
+        mMsg_request_main_forceoff();
+        aFLAG_setup_action(flag, aFLAG_ACTION_WAIT);
+    }
+}
+
+static void aFLAG_talk(STRUCTURE_ACTOR* flag, GAME_PLAY* game_play) {
+    mMsg_Window_c* msg_p;
+    PLAYER_ACTOR* player_actor;
+    s16 target_y;
+    s_xyz target_rot;
+
+    msg_p = mMsg_Get_base_window_p();
+    if (mDemo_Check(mDemo_TYPE_TALK, (ACTOR*)flag) != FALSE) {
+        player_actor = get_player_actor_withoutCheck(game_play);
+
+        target_rot = player_actor->actor_class.shape_info.rotation;
+        target_y = search_position_angleY(&player_actor->actor_class.world.position, &flag->actor_class.world.position);
+        add_calc_short_angle2(&target_rot.y, target_y, 0.3f, 0x1000, 0x100);
+
+        (*get_player_actor_withoutCheck((GAME_PLAY*)gamePT)->Set_force_position_angle_proc)(
+            gamePT, NULL, &target_rot, mPlayer_FORCE_POSITION_ANGLE_ROTY);
+
+        if (mMsg_Check_MainNormalContinue(msg_p) == TRUE) {
+            switch (mChoice_Get_ChoseNum(mChoice_Get_base_window_p())) {
+                case mChoice_CHOICE0:
+                    mMsg_request_main_disappear_wait_type1(mMsg_Get_base_window_p());
+                    aFLAG_setup_action(flag, aFLAG_ACTION_DOWN);
+                    break;
+
+                case mChoice_CHOICE1:
+                    mMsg_request_main_disappear_wait_type1(mMsg_Get_base_window_p());
+                    aFLAG_setup_action(flag, aFLAG_ACTION_TALK_END);
+                    break;
+            }
+        }
+    } else {
+        aFLAG_setup_action(flag, aFLAG_ACTION_WAIT);
+    }
+}
+
+static void aFLAG_wait(STRUCTURE_ACTOR* flag, GAME_PLAY* game_play) {
+    PLAYER_ACTOR* player_actor;
+    if (mDemo_Check(mDemo_TYPE_TALK, (ACTOR*)flag) == FALSE) {
+        player_actor = get_player_actor_withoutCheck(game_play);
+        if (player_actor != NULL) {
+            mDemo_Request(mDemo_TYPE_TALK, (ACTOR*)flag, aFLAG_set_talk_info);
+        }
+    } else {
+        aFLAG_setup_action(flag, aFLAG_ACTION_TALK);
+    }
+}
+
+static void aFLAG_menu_open_wait(STRUCTURE_ACTOR* flag, GAME_PLAY* game_play) {
+    mSM_open_submenu(&game_play->submenu, mSM_OVL_NEEDLEWORK, 0, common_data.player_no);
+    aFLAG_setup_action(flag, aFLAG_ACTION_END_WAIT);
+}
+
+static void aFLAG_menu_end_wait(STRUCTURE_ACTOR* flag, GAME_PLAY* game_play) {
+    Submenu* submenu;
+    Submenu_Item_c* sm_item;
+    u8 design_index;
+
+    submenu = &game_play->submenu;
+    if (submenu->open_flag == FALSE) {
+        sm_item = submenu->item_p;
+        if (sm_item->item == RSV_NO) {
+            design_index = mNW_get_image_no(submenu, sm_item->slot_no) & 7;
+
+            bcopy(Player_Design_Get(design_index), Save_GetPointer(island.flag_design.design),
+                  sizeof(mNW_original_tex_c));
+            osWritebackDCache(Save_GetPointer(island.flag_design.design), sizeof(mNW_original_tex_c));
+
+            Save_Get(island.flag_design)._11 = TRUE;
```

Comment:
There might be a union type for designs since this TU seems to be the only place where this member is accessed. Regardless, we can probably leave it as a struct.

I'd like to give this member a name before merging this PR. Maybe something like 'flag_design_set' would be appropriate? Feel free to pick another name if you want though!